### PR TITLE
Add development mode to lab extension build scripts

### PR DESCRIPTION
**Pull request by jasongrout**
## References

This is built on #8913, so that should be merged first.

## Code changes

This adds a development option to lab extension building scripts. The default is to build in production mode.

Also renames `static_path` option to `static_url`.

### CODE REVIEW

**Comment by blink1073** [comment]
Why not just merge this one since the rebase is done and build is in progress?

**Comment by blink1073** [comment]
Ah, I see you already rebased

**Comment by blink1073** [comment]
I pushed fixes for `examples` and `usage` failures

**Comment by blink1073** [comment]
`browser_check` is failing with a dynamic extension:

```
[W 2020-08-27 14:36:28.825 ServerApp] '_build'
...
[W 2020-08-27 14:36:40.742 ServerApp] 404 GET /foo/lab?token=fa278731aea392936cec7c943c946ab5600f2e988fab7c37 (::1) 34.89ms referer=None
```

I think there is a problem looking up the `_build` metadata.

**Comment by blink1073** [comment]
The `usage` failure seems to have snuck in on https://github.com/jupyterlab/jupyterlab/pull/8913

**Comment by blink1073** [comment]
With 0e5be7329 we get `404 GET /foo/lab/extensions/@jupyterlab/mock-extension/remoteEntry.js`, which is expected until we make a release.

**Comment by jasongrout** [comment]
Two other things when I did this. I'd love your thoughts:

1. "static path" seems like a confusing name for an option, especially next to "core path". Really, it's a URL, most likely including the host, not a path on disk like the core path is.
2. (an implementation detail...) - I think we should pass values from build-labextension into the webpack config using the webpack cli env feature, rather than relying on environment variables.

File: jupyterlab/labapp.py [hunk 1]
```diff
@@ -709,7 +709,7 @@ def initialize_handlers(self):
         info = get_app_info()
         extensions = page_config['dynamic_extensions'] = []
         for (ext, ext_data) in info.get('dynamic_exts', dict()).items():
-            extbuild = ext_data['_build']
+            extbuild = ext_data['jupyterlab']['_build']
```
Comment (jasongrout) [inline jupyterlab/labapp.py:712]: Thanks!

**Comment by blink1073** [comment]
> "static path" seems like a confusing name for an option, especially next to "core path". Really, it's a URL, most likely including the host, not a path on disk like the core path is.

Sure, `static_url` seems better, I'll update.

>  (an implementation detail...) - I think we should pass values from build-labextension into the webpack config using the webpack cli env feature, rather than relying on environment variables.

Makes sense to me.

**Comment by jasongrout** [comment]
Thanks for the cleanups!

File: builder/src/build-labextension.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ commander
   .description('Build an extension')
   .option('--development', 'build in development mode')
   .requiredOption('--core-path <path>', 'the core package directory')
-  .option('--static-path <path>', 'static path for build assets')
+  .option('--static-url <url>', 'static path for build assets')
```
Comment (jasongrout) [inline builder/src/build-labextension.ts:25]:
```suggestion
  .option('--static-url <url>', 'url for build assets')
```

**Pull request merged**